### PR TITLE
Migrated TC test_self_heal_with_meta_data_entry_and_files_removed.py

### PR DESCRIPTION
**Pull request by Sheetalpamecha**
[Migrated TC test_self_heal_with_meta_data_entry_and_files_removed.py](https://github.com/gluster/glusto-tests/blob/35f1b97a8ecee852b2543e87ceb2ebbe9cf4f6ca/tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py)

Updates:#292
Signed-off-by: Sheetal Pamecha <spamecha@redhat.com>

### CODE REVIEW

**Comment by Sheetalpamecha** [comment]
```
✔ Running test case : test_self_heal_with_meta_data_entry_and_files_removed-dist-rep
✔ test_self_heal_with_meta_data_entry_and_files_removed-dist-rep Succeeded
✔ Running test case : test_self_heal_with_meta_data_entry_and_files_removed-rep
✔ test_self_heal_with_meta_data_entry_and_files_removed-rep Succeeded
Results for heal
Test nature : disruptive
TestName : test_self_heal_with_meta_data_entry_and_files_removed
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|   dist-rep  |     PASS    |        00:20:51       |      NA     |
|     rep     |     PASS    |        00:21:02       |      NA     |
+-------------+-------------+-----------------------+-------------+
+--------------------------------------+-------+
|               Category               | Count |
+--------------------------------------+-------+
|           Disruptive Tests           |   1   |
|         Non Disruptive Tests         |   0   |
|       Disruptive Tests Skipped       |   0   |
|     Non Disruptive Tests Skipped     |   0   |
|              Tests Ran               |   2   |
|            Tests Skipped             |   0   |
|             Total Tests              |   1   |
|   Disruptive Tests Pass Percentage   | 100.0 |
| Non Disruptive Tests Pass Percentage |   0   |
|        Total Pass Percentage         | 100.0 |
+--------------------------------------+-------+
Total Time Taken : 00 hours 42 minutes 22 seconds
✔ Tearing down successful.
```

File: tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+# disruptive;dist-rep,rep
+import traceback
+from tests.d_parent_test import DParentTest
+
+
+class TestCase(DParentTest):
+
+    def terminate(self):
+        # Delete non-root users and group if created
+        if self.user_group_created:
```
Comment (nik-redhat) [inline tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py:31]: This should be under the try block

File: tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+                    raise Exception(f"Unable to delete group 'qa_all'"
+                                    f"on {self.client_list[0]}")
+            except Exception as error:
+                tb = traceback.format_exc()
+                self.redant.logger.error(error)
+                self.redant.logger.error(tb)
+            super().terminate()
+
+    def _setup_user(self):
+        self.user_group_created = False
```
Comment (nik-redhat) [inline tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py:52]: This should be defined at the beginning of the `run_test` function, as before reaching this definition the TC might fail

**Review by nik-redhat — CHANGES_REQUESTED** [review]

File: tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py (hunk 2 above)
Comment (Sheetalpamecha) [reply tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py]: Done

File: tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py (hunk 1 above)
Comment (Sheetalpamecha) [reply tests/functional/afr/heal/test_self_heal_with_meta_data_entry_and_files_removed.py]: Done

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**